gr-filter/grc/variable_band_pass_filter_taps.block.yml

```diff
@@ -36,6 +36,9 @@ parameters:
     label: Beta
     dtype: float
     default: '6.76'
+-   id: value
```

Comment:
What is this additional parameter used for?

Reply:
Without this entry cpp generation fails with:

  File "/usr/local/gnuradio/lib64/python3.10/site-packages/gnuradio/grc/core/generator/cpp_top_block.py", line 86, in write
    self._variable_types()
  File "/usr/local/gnuradio/lib64/python3.10/site-packages/gnuradio/grc/core/generator/cpp_top_block.py", line 310, in _variable_types
    if var.params['value'].dtype != 'raw':
KeyError: 'value'

Reply:
I checked some other blocks that claimed to be a variable block and to be able to generate cpp code.
If there is no value entry cpp code generation fails. Just try the Default Header Format Obj.
We are talking about blocks of 'type variable'. A variable represents a value, so the value parameter is not a dummy parameter, but more important for the cpp generation, it's associated with a type, and this type is used for cpp code generation. This is important. Take the above mentioned filter tap example and enable the len_taps block. Then you can't generate cpp code anymore, as deriviation of the type from the expression len(lp_taps) fails for cpp, though the error message is misleading.
So I believe, it's the right way to add the value parameter.
At the moment there are lot of cpp templates for variable generation, that fail, but could be fixed in this way. But apparently they are not used.

Reply:
Ok, I see those now - I guess it makes sense as the way it is worked around.